Dockerfile

```diff
@@ -8,7 +8,9 @@ RUN go mod download
 COPY cmd cmd
 COPY internal internal
 COPY apis apis
-RUN CGO_ENABLED=0 GOOS=linux GOFLAGS=-ldflags=-w go build -o /go/bin/contour -ldflags=-s -v github.com/projectcontour/contour/cmd/contour
+
+ARG CONTOUR_VERSION
+RUN CGO_ENABLED=0 GOOS=linux GOFLAGS=-ldflags=-w go build -ldflags="-s -X github.com/projectcontour/contour/internal/build.Version=${CONTOUR_VERSION}" -o /go/bin/contour -v github.com/projectcontour/contour/cmd/contour
```

Comment:
Conventionally, the build info is picked up in a Makefile at build time:

For example:
```
SHA := $(shell git rev-parse HEAD)
REVISION := $(shell git describe --tags --exact-match HEAD)

build: 
	@$(GO) build \
		-ldflags "-X $(REPO)/pkg/version.Revision=$(REVISION)" \
		-ldflags "-X $(REPO)/pkg/version.Sha=$(SHA)" \
		-o $(BIN) .
```

This lets you have exactly one place with the magic incantations. The variables from the makefile can be passed through the docker build, and then inside the docker file, just run "make build" (after adding the "build" target).

Reply:
Experiment with the git commands a bit; the version for release builds should always be a tag name, but for working trees, you probably want the short hash or something.